### PR TITLE
feat(status): include bee mode in response

### LINKED ISSUE #3977: status api refactor

### PR DESCRIPTION
**Pull request by mrekucci**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [x] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
include bee mode in the snapshot response; refactoring and cleanup

Closes #3977

### CODE REVIEW

File: pkg/status/internal/pb/status.proto [hunk 1]
```diff
@@ -15,9 +15,10 @@ message Get {}
 // the appropriate values that are a snapshot of the current state
 // of the running node.
 message Snapshot {
-  uint64 ReserveSize = 1;
-  double PullsyncRate = 2;
-  uint32 StorageRadius = 3;
-  uint64 ConnectedPeers = 4;
-  uint64 NeighborhoodSize = 5;
+  string BeeMode = 1;
```
Comment (aloknerurkar) [inline pkg/status/internal/pb/status.proto:18]: If we want to maintain backward compatibility, we should be adding this field at the end I think. This allows us to handle messages from older protocol as well.

Comment (mrekucci) [reply pkg/status/internal/pb/status.proto:18]: So the minor version of the protocol is bumped to cover that.

**Review by istae — COMMENTED** [review]
Can we added the same fields from `/status/peer` to `/status` as well pls?

**Comment by mrekucci** [comment]
> Can we add the same fields from `/status/peer` to `/status` as well pls?

It is already done in this PR.

**Review by aloknerurkar — APPROVED** [review]

**Review by istae — APPROVED** [review]

**Pull request merged**